[BUG] Failure to upgrade JVector version to rc8

**What is the bug?**
Upgrading the project fails with the compilation error, due to API changes on the `PQVectors.encodeAndBuild` method: 

```
/Users/akashshankaran/Opensearch/opensearch-jvector/src/main/java/org/opensearch/knn/index/codec/jvector/JVectorWriter.java:962: error: incompatible types: int[] cannot be converted to IntUnaryOperator
                    compactOrdsToRavvOrds,
                    ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error
5 warnings
```

**How can one reproduce the bug?**
Steps to reproduce the behavior:
1. Upgrade version of Jvector library to 4.0.0-rc.8
2. Run `./gradlew build`
3. Scroll down to '....'
4. See error

**What is the expected behavior?**
Code should compile, and all tests should pass

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 * Fixing guava dependency scope, since the dependency is provided by transport-grpc plugin [344] (https://github.com/opensearch-project/opensearch-jvector/pull/344)
 ### Infrastructure
 * Upgrade Lucene to 10.4.0 [292] (https://github.com/opensearch-project/opensearch-jvector/pull/292)
+* Upgrade jvector from 4.0.0-rc.6 to 4.0.0-rc.8 [370](https://github.com/opensearch-project/opensearch-jvector/pull/370)
 ### Documentation
 ### Maintenance
 * Update `com.google.guava:failureaccess` from 1.0.1 to 1.0.2

--- a/gradle.properties
+++ b/gradle.properties
@@ -5,7 +5,7 @@
 
 version=1.0.0
 systemProp.bwc.version=3.2.0
-jvector_version=4.0.0-rc.6
+jvector_version=4.0.0-rc.8
 java_release_version=21
 
 # org.gradle.jvmargs=--add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \

--- a/src/main/java/org/opensearch/knn/index/codec/jvector/JVectorWriter.java
+++ b/src/main/java/org/opensearch/knn/index/codec/jvector/JVectorWriter.java
@@ -961,15 +961,7 @@ public class JVectorWriter extends KnnVectorsWriter {
                 final long trainingTime = end - start;
                 log.info("Refined PQ codebooks for field {}, in {} millis", fieldName, trainingTime);
                 KNNCounter.KNN_QUANTIZATION_TRAINING_TIME.add(trainingTime);
-                compactPqVectors = PQVectors.encodeAndBuild(
-                    leadingCompressor,
-                    // graphNodeIdsToRavvOrds.length,
-                    // graphNodeIdsToRavvOrds,
-                    compactOrdsToRavvOrds.length,
-                    compactOrdsToRavvOrds,
-                    this,
-                    simdPoolMerge
-                );
+                compactPqVectors = PQVectors.encodeAndBuild(leadingCompressor, compactRavv.size(), compactRavv, simdPoolMerge);
             }
 
             if (compactPqVectors == null) {

